src/moscot/base/output.py

```diff
@@ -168,6 +171,93 @@ def chain(self, outputs: Iterable["BaseSolverOutput"], scale_by_marginals: bool
 
         return op
 
+    def sparsify(
+        self,
+        mode: Literal["threshold", "percentile", "min_row"],
+        threshold: Optional[float] = None,
+        batch_size: int = 1024,
+        n_samples: Optional[int] = None,
+        seed: Optional[int] = None,
+    ) -> None:
+        """
+        Sparsify the :attr:transport_matrix.
+
+        This function sets all entries of the transport matrix below `threshold` to 0 and
+        returns the result as a :class:`~scipy.sparse.csr_matrix`.
+
+        Parameters
+        ----------
+        mode
+            Which threshold to use for sparsification. Can be one of:
+
+                - 'threshold' - threshold below which entries are set to 0.0.
+                - 'percentile' - determine threshold by percentile below which entries are set to 0. Hence, between 0
+                  and 100.
+                - 'min_row' - choose the threshold such that each row has at least one non-zero entry.
+
+        threshold
+            Threshold or percentile depending on ``mode```. If ``mode = 'min_row'``, ``threshold``` can be neglected.
+        batch_size
+            How many rows of the transport matrix to sparsify per batch.
+        n_samples
+            If ``mode = 'percentile'``, determine the number of samples based on which the percentile
+            is computed stochastically. Note this means that a matrix of shape `[n_samples, transport_matrix.shape[1]]`
+            has to be instantiated. If `None`, ``n_samples`` is set to batch_size.
+        seed
+            Random seed needed for sampling if ``mode = 'percentile'``.
+
+        Returns
+        -------
+        Nothing, but adds the sparsified transport matrix to :attr:`self.sparsified_tmap`.
+
+        Note
+        ----
+        This function only serves for interfacing software which has to instantiate the transport matrix. Within
+        :mod:`moscot`, there is no point in using this function. We encourage users not to use this function unless it
```

Comment:
I think this comment can scare the user, esp. the last sentence. Would soften this a bit.